File: ruby/raindrops/raindrops.rb
class Raindrops
  def self.convert(num)
    res = ""
    res << if num % 3 == 0 then "Pling" else "" end
    res << if num % 5 == 0 then "Plang" else "" end
    res << if num % 7 == 0 then "Plong" else "" end
    if res.empty?
      res = num.to_s
    end
    return res
  end
end

File: r/raindrops/raindrops.R
raindrops <- function(number) {
  result <- ""
  
  if (number %% 3 == 0) {
    result <- paste0(result, "Pling")
  }
  if (number %% 5 == 0) {
    result <- paste0(result, "Plang")
  }
  if (number %% 7 == 0) {
    result <- paste0(result, "Plong")
  }
  
  if (result == "") {
    result <- as.character(number)
  }
  
  return(result)    
}
